plaidml/op/lib/ops.cc

```diff
@@ -2914,6 +2915,53 @@ Value transpose(const Value& value) {
   return Value{O};
 }
 
+Value unsqueeze(const Value& value) {
+  // Read arguments
+  auto args = value.as_tuple();
+  if (args.size() != 2) {
+    throw std::runtime_error(llvm::formatv("PlaidML unsqueeze op expects 2 arguments (received {0})", args.size()));
+  }
+
+  auto I = args[0].as_tensor();
+  auto ndims = I.rank();
+
+  std::vector<int64_t> raw_axes;
+  if (args[1].is_int()) {
+    raw_axes.push_back(args[1].as_int());
+  } else {
+    raw_axes = args[1].as_int_tuple();
+  }
+  std::set<size_t> axes;
+  for (auto& raw_axis : raw_axes) {
+    axes.insert(normalize_axis(raw_axis, ndims + 1, "unsqueeze"));
```

Comment:
This should be `ndims + axes.size()`: The axes numbers are relative to the output tensor, which has as many more dimensions than `I` as the size of the `axes` set.